integrations/ad_integration/ad_reader.py

```diff
@@ -117,32 +128,40 @@ def _get_from_ad(self, user=None, cpr=None, school=False):
             # get more users at the same time to increase performance.
             # Lookup time is only very slightly dependant on the number
             # of results.
-            ps_template = "get-aduser -Filter 'xAttrCPR -like \"{}\"'"
+            field = settings['cpr_field']
+            ps_template = "get-aduser -Filter '" + field + " -like \"{}\"'"
 
         get_command = ps_template.format(dict_key)
 
-        if school:
-            search_base = ('-Server uv-viborg.local ' +
-                           '-SearchBase "DC=uv-viborg,DC=local" ')
-        else:
-            search_base = ' -SearchBase "OU=Kommune,DC=viborg,DC=local" '
+        server = ''
+        if settings['server']:
+            server = ' -Server {} '.format(settings['server'])
 
+        search_base = ' -SearchBase "{}" '.format(settings['search_base'])
         credentials = ' -Credential $usercredential'
-        if school:
-            properties = (' | Get-ADObject -Properties xAttrCPR,ObjectGuid,' +
-                          'SamAccountName,Title,Name,mail')
-        else:
-            # properties = ' -Properties *'
-            properties = (' -Properties xAttrCPR,ObjectGuid,SamAccountName,Title,' +
-                          'Name,xBrugertype,EmailAddress, MobilePhone')
+
+        get_ad_object = ''
+        if settings['get_ad_object']:
+            get_ad_object = ' | Get-ADObject'
+
+        # properties = ' -Properties *'
+        properties = ' -Properties '
+        for item in settings['properties']:
+            properties += item + ','
+        properties = properties[:-1] + ' '  # Remove trailing comma, add space
```

Comment:
Ikke for at være en smartass, og jeg kan sagtens se at det virker, da jeg lige havde læst det et par gange :-)
Jeg kunne blot bedre ønske mig noget som:
```
properties = ' -Properties {} '.format(",".join(settings["properties"]))
```
Men når jeg nu kigger på det er det måske ikke meget klarere

Reply:
Det afhænger nok at øjnene der ser, men jeg synes ikke det er lettere at læse :)